### PR TITLE
Allow specification of the FLX bundle (containing no Dart snapshot) outside the iOS/Mac application bundle.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Allows launching of apps without having any Xcodebuild step in the simulator. Relaunch the application in either the iOS simulator or the Mac shell to see the latest changes.

### CODE REVIEW

File: sky/shell/platform/mac/sky_window.mm [hunk 1]
```diff
@@ -100,15 +102,10 @@ - (void)setupAndLoadDart {
...
-        command_line.GetSwitchValueASCII(sky::shell::switches::kPackageRoot));
-    return;
-  }
-
-  NSString *endpoint = self.skyInitialBundleURL;
-  if (endpoint.length > 0) {
-    mojo::String string(endpoint.UTF8String);
-    _sky_engine->RunFromBundle(string);
+    _sky_engine->RunFromFile(
+        args[0],  //
```
Comment (abarth) [inline sky/shell/platform/mac/sky_window.mm]: `//`  ?

**Comment by chinmaygarde** [comment]
* Removes xcodebuild from the simulator turnaround cycle. Relaunch the app (in either the iOS simulator or Mac shell) to see latest changes.
* Does not generate the Dart snapshot that needs to be stuffed into the FLX.  This was the biggest cause of the having to repackage the application bundle (and incur the time to repackage the same).
* FLX regeneration for assets does not require codesigning. Less frequent but still required repackaging. No more.

File: sky/shell/ui/engine.cc [hunk 2]
```diff
@@ -203,8 +202,13 @@ void Engine::RunFromPrecompiledSnapshot(const mojo::String& bundle_path) {
 }
 
 void Engine::RunFromFile(const mojo::String& main,
-                         const mojo::String& package_root) {
+                         const mojo::String& package_root,
+                         const mojo::String& bundle) {
   TRACE_EVENT0("flutter", "Engine::RunFromFile");
+  if (bundle.size() != 0) {
```
Comment (abarth) [inline sky/shell/ui/engine.cc]: if (!bundle.empty()) { ...

**Comment by abarth** [comment]
LGTM

File: sky/shell/platform/mac/sky_window.mm (hunk 1 above)
Comment (chinmaygarde) [inline sky/shell/platform/mac/sky_window.mm]: To fool clang-format into not indenting the arguments in a weird way. It usually does an extremely good job. But in this case, it made it look like the last argument to RunFromFile was an argument to GetSwitchValueASCII.

Comment (abarth) [inline sky/shell/platform/mac/sky_window.mm]: Maybe capture the result of GetSwitchValueASCII into a local to make the line format cleanly?  That's probably better than having a stray comment.  :)

Comment (chinmaygarde) [inline sky/shell/platform/mac/sky_window.mm]: Done. Rebased.

**Pull request merged**